'test_blob_upload_w_bucket_cmek_enabled' systest flakes

From [this Kokoro failure](https://source.cloud.google.com/results/invocations/bccf9f4d-ad47-480a-9e6a-fcc64e5a62f5/targets/cloud-devrel%2Fclient-libraries%2Fpython%2Fgoogleapis%2Fpython-storage%2Fpresubmit%2Fpresubmit/log):

```
____________________ test_blob_upload_w_bucket_cmek_enabled ____________________

kms_bucket = <Bucket: gcp-systest-kms-1626804473143>
blobs_to_delete = [<Blob: gcp-systest-kms-1626804473143, test-blob, 1626804479031944>]
kms_key_name = 'projects/precise-truck-742/locations/us/keyRings/gcs-test/cryptoKeys/gcs-test'
file_data = {'big': {'hash': 'cEome4a+NYd7YIXzXQnR5Q==', 'path': '/tmpfs/src/github/python-storage/tests/data/five-point-one-mb-fi...ng'}, 'simple': {'hash': '3Hkwjv2WvCnKjNR6Z3CboA==', 'path': '/tmpfs/src/github/python-storage/tests/data/simple.txt'}}

    def test_blob_upload_w_bucket_cmek_enabled(
        kms_bucket, blobs_to_delete, kms_key_name, file_data,
    ):
        blob_name = "test-blob"
        payload = b"DEADBEEF"
        alt_payload = b"NEWDEADBEEF"

        kms_bucket.default_kms_key_name = kms_key_name
        kms_bucket.patch()
        assert kms_bucket.default_kms_key_name == kms_key_name

        blob = kms_bucket.blob(blob_name)
        blob.upload_from_string(payload)
        blobs_to_delete.append(blob)

        _helpers.retry_429_harder(blob.reload)()
        # We don't know the current version of the key.
>       assert blob.kms_key_name.startswith(kms_key_name)
E       AttributeError: 'NoneType' object has no attribute 'startswith'
```

--- a/tests/system/_helpers.py
+++ b/tests/system/_helpers.py
@@ -54,8 +54,13 @@ def _no_event_based_hold(blob):
     return not blob.event_based_hold
 
 
+def _has_kms_key_name(blob):
+    return blob.kms_key_name is not None
+
+
 retry_bad_copy = RetryErrors(exceptions.BadRequest, error_predicate=_bad_copy)
 retry_no_event_based_hold = RetryInstanceState(_no_event_based_hold)
+retry_has_kms_key_name = RetryInstanceState(_has_kms_key_name)
 
 
 def unique_name(prefix):

--- a/tests/system/test_kms_integration.py
+++ b/tests/system/test_kms_integration.py
@@ -229,7 +229,7 @@ def test_blob_upload_w_bucket_cmek_enabled(
     blob.upload_from_string(payload)
     blobs_to_delete.append(blob)
 
-    _helpers.retry_429_harder(blob.reload)()
+    _helpers.retry_429_harder(_helpers.retry_has_kms_key_name(blob.reload))()
     # We don't know the current version of the key.
     assert blob.kms_key_name.startswith(kms_key_name)
 

